src/api/controllers/users.js

```diff
@@ -99,4 +99,78 @@ function patchProfileByAuthUser(req, res) {
     }
 }
 
-export default { getUsers, getUserByAuthUser, getUserById, patchProfileByAuthUser } 
\ No newline at end of file
+// GET /users/auth/favorites
+// student only
+function getFavoritesByAuthUser(req, res) {
+    if (req.user.user_type !== 'student') {
+        return res.status(403).json({
+            "message": response.onlyStudent
+        });
+    } else {
+        try {
+            User.findById(req.user._id).populate('favorites').exec(function(err, user) {
+                return res.status(200).json({
```

Comment:
is there a benefit of try catch block over doing the if (err) check inside of the exec block?

just curious

Reply:
I don't remember. When I first wrote it I just had an outer try/catch, but that didn't actually catch some rejected promises inside the query block. So then I added the inner try/catch. It is probably best to just keep it consistent with what we had before, which is an inner try/catch block for the inner logic, and and if(err) check to catch outer query errors. 

I will change these to remove the outer try/catch and add if(err) checks